### DISCUSSION TITLE
Bulk Operations and Performance Benchmarks Inquiry

### DISCUSSION THREAD

**Discussion by uazapi** [pocketbase/pocketbase#4503] [Q&A] [answered]
Hello ganigeorgiev,

Firstly, I would like to express how much I admire PocketBase. The simplicity and ease of use it offers have been fundamental in the development of my projects.

However, as I plan for the scalability of my project, some important questions have arisen:

Bulk Operations via API: I've noticed that PocketBase currently does not support bulk insertions and updates through the API. Such functionalities are crucial for the efficient handling of large volumes of data. Are there plans to include these capabilities in future versions?

Data Import/Export: The lack of integrated functionalities for data import and export leads me to consider alternative solutions, like custom scripts. A graphical interface for these operations would greatly facilitate data management, especially for non-developer users. Is there any forecast for the development of a GUI that enables data import and export?

SQLite Performance Benchmark: In a previous discussion, you mentioned that SQLite, in your evaluations, proved to be faster than Postgres for certain cases. This statement piqued my interest, especially when considering the performance needs of my application. Could you share benchmarks or comparative data that support this observation?

Depending on this information, I am considering developing scripts or custom APIs for bulk operations, despite recognizing that adding these functions to the admin-ui may be beyond my current technical knowledge. And regarding the benchmarks, it will also help me decide whether I should continue using SQLite or consider forking the project to use Postgres.

I appreciate your attention and understanding in advance. Your work on PocketBase is truly inspiring, and I am eager to follow the platform's evolution and how it can better meet the needs of developers and businesses around the world.

Best regards,

**Comment by ganigeorgiev** [comment, marked as answer]
Bulk/transaction web APIs are currently on hold but will be available eventually in the future after the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) (there are no ETAs).

Data Import/Export is tracked in https://github.com/pocketbase/pocketbase/issues/48 and it is also on hold as it relies on the bulk/transaction web APIs support.

For the SQLite Performance Benchmark - if you are looking for raw SQLite vs PostgreSQL benchmarks you 
could check https://www.golang.dk/articles/benchmarking-sqlite-performance-in-go.

For PocketBase related benchmarks you can explore the results of the benchmarks repo - https://github.com/pocketbase/benchmarks.

For some numbers regarding a production PocketBase project you could check https://github.com/pocketbase/pocketbase/discussions/4254.

> And regarding the benchmarks, it will also help me decide whether I should continue using SQLite or consider forking the project to use Postgres.

Using PocketBase with PostgreSQL will be always slower in terms of pure query execution, especially for read queries and if the PostgreSQL database is hosted on a different server.

If your use case cannot be managed by a single server, then PocketBase is not suitable for it.
